src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java

```diff
@@ -372,6 +381,110 @@ public void testGetSnapshotTableData() throws Exception {
+                        .datatype(TableDataType.FILEREF)
+                        .arrayOf(true)));
+
+    Dataset dataset = readDataset(datasetRequest);
+
+    // Stage tabular data for ingest.
+    String targetPath = "scratch/file" + UUID.randomUUID() + "/";
+    String bucket = testConfig.getIngestbucket();
+    BlobInfo fileBlob = BlobInfo.newBuilder(bucket, targetPath + "ingest-test-file.json").build();
+
+    try {
+      assertThatDatasetAndTablesShouldExist(dataset, false);
+
+      bigQueryDatasetPdao.createDataset(dataset);
+      assertThatDatasetAndTablesShouldExist(dataset, true);
+
+      //  ingest data into the new dataset.
+      UUID newFileId1 = UUID.randomUUID();
+      UUID newFileId2 = UUID.randomUUID();
+      String fileIngestDataTmpl1 =
+          "{\"sourcePath\":\"%s\", \"targetPath\":\"/file/%s\"}".formatted(gsPath(fileBlob), "0");
+      String fileIngestDataTmpl2 =
+          "{\"sourcePath\":\"%s\", \"targetPath\":\"/file/%s\"}".formatted(gsPath(fileBlob), "1");
+      String ingestData =
+          """
+        {"id":"file1","derived_from":["sample6","sample2"],"file":%s,"file_a":[%s]}
+      """
+              .formatted(fileIngestDataTmpl1, fileIngestDataTmpl2);
+      storage.create(fileBlob, ingestData.getBytes(StandardCharsets.UTF_8));
+      IngestRequestModel ingestRequest =
+          new IngestRequestModel().format(IngestRequestModel.FormatEnum.JSON);
+      connectedOperations.ingestTableSuccess(
+          dataset.getId(), ingestRequest.table("file").path(gsPath(fileBlob)));
+      List<String> originalIds = readIds(dataset);
+
+      // Insert the Id mappings
+      // Note: we need a transaction to power the live view used as part of inserting new file ids
+      String flightId = "myflight";
+      TransactionModel transaction =
+          bigQueryTransactionPdao.insertIntoTransactionTable(TEST_USER, dataset, flightId, null);
+      bigQueryDatasetPdao.createStagingFileIdMappingTable(dataset);
+      bigQueryDatasetPdao.fileIdMappingToStagingTable(
+          dataset,
+          Map.of(
+              UUID.fromString(originalIds.get(0)), newFileId1,
+              UUID.fromString(originalIds.get(1)), newFileId2));
+
+      // Update the metadata table
+      bigQueryDatasetPdao.insertNewFileIdsIntoDatasetTable(
+          dataset, getTable(dataset, "file"), transaction.getId(), flightId, TEST_USER);
+      // Commit the transaction
+      bigQueryTransactionPdao.updateTransactionTableStatus(
+          TEST_USER, dataset, transaction.getId(), StatusEnum.COMMITTED);
+
+      List<String> newIds = readIds(dataset);
+      assertThat(
+          "new file ids are now stored",
+          newIds,
+          containsInAnyOrder(newFileId1.toString(), newFileId2.toString()));
+
+      // Now delete it and test that it is gone
+      bigQueryDatasetPdao.deleteDataset(dataset);
+      assertThatDatasetAndTablesShouldExist(dataset, false);
+    } finally {
+      datasetDao.delete(dataset.getId(), TEST_USER);
```

Comment:
legacy tests...I started going through this one and it's a bit of a weird one (it partially creates datasets, I think to speed up the test.  I'm gonna leave this one alone for now since it's gonna be a major refactor of this class

Reply:
No problem! Makes sense to cal that OOS